lib/AST/Decl.cpp

```diff
@@ -2835,7 +2835,14 @@ OpaqueReturnTypeRepr *ValueDecl::getOpaqueResultTypeRepr() const {
     returnRepr = SD->getElementTypeRepr();
   }
 
-  return dyn_cast_or_null<OpaqueReturnTypeRepr>(returnRepr);
+  // TODO [OPAQUE SUPPORT]: once we want to actually perform semantic analysis
+  // on `NamedOpaqueReturnTypeRepr`s, we will need to also check `||
+  // isa<NamedOpaqueReturnTypeRepr>` here.
+  return returnRepr && returnRepr->findIf([](TypeRepr *ty) {
+    return isa<OpaqueReturnTypeRepr>(ty);
+  })
+             ? returnRepr
```

Comment:
This combined && and ternary expression is hard to read. Please write it out on multiple lines using ifs